src/components/include/cos_thd_init.h

```diff
@@ -20,6 +20,14 @@ __init_data_alloc(void *fn, void *data) {
 
 			assert(__thd_init_data[i].fn == fn);
 			__thd_init_data[i].data = data;
+			__thd_init_data[i].type = type;
+			if (type == THD_AEP) {
```

Comment:
I think you can get rid of the type stuff if you use my solution that doesn't modify existing code.